core/frontend/src/request/utils.ts

```diff
@@ -0,0 +1,52 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import { Base64EncodedString } from "@itwin/core-common";
+import { RequestBasicCredentials } from "./Request";
+
+/**
+ * Check whether or not one of the requested authentication method is listed in the HTTP 'WWW-Authenticate' response header
+ * @param headers Headers object
+ * @param query List of authentication method to lookup
+ * @note For CORS requests, the 'Access-Control-Expose-Headers' header from the server must make the 'WWW-Authenticate' available to the browser, otherwise this won't work.
+ * @internal
+ */
+export function headersIncludeAuthMethod(headers: Headers, query: string[]): boolean {
+  const headersArray = Array.from(headers);
+  const foundAuth = headersArray.filter((pair)=> pair[0].toLowerCase() === "www-authenticate");
```

Comment:
why not just use [Headers.has](https://developer.mozilla.org/en-US/docs/Web/API/Headers/has) and [Headers.get](https://developer.mozilla.org/en-US/docs/Web/API/Headers/get)

Reply:
For some reasons I assumed has/get were case sensitive and I wanted to avoid any case related issues .  
After testing, looks like those methods are already [case-insensitive](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAFgUwIYBMECcIwLwzAgdxgAlk1MAKASgG4AoRVDCAOggSgoHIB1PgWgCCAVyiIwUAJbAkUBFwA0MLgDkAKgBkAsl1p1QkEABsELIyADmFAAbEkWXgJFiEE6bPkAuGABIA3ozkrHD2NgTh-Eii4lIyctZUAL4JNEA), I will change the code.

Reply:
should be better now.